<?xml version="1.0" encoding="utf-8"?>
<launch>
<arg name="base" value="$(optenv TURTLEBOT_BASE kobuki)"/> <!-- create, roomba -->
<arg name="stacks" value="$(optenv TURTLEBOT_STACKS hexagons)"/>  <!-- circles, hexagons --> 
<arg name="3d_sensor" value="$(optenv TURTLEBOT_3D_SENSOR kinect)"/>  <!-- kinect, asus_xtion_pro --> 
<arg name="urdf_file" default="$(find xacro)/xacro.py '$(find turtlebot_description)/robots/$(arg base)_$(arg stacks)_$(arg 3d_sensor).urdf.xacro'"/>
<param name="robot_description" command="$(arg urdf_file)"/>


<!-- Begin robot 1 -->
<group ns ="robot1">
  <param name = "tf_prefix" value ="robot1_tf"/>
  <include file = "$(find path_generator)/launch/one_robot.launch">
  <arg name= "init_pose" value="-x -1 -y -1 -z 0" />
  <arg name= "robot_name" value="robot1" />
  </include> 
</group>

<!-- Begin robot 2 -->
<group ns ="robot2">
  <param name = "tf_prefix" value ="robot2_tf"/>
  <include file = "$(find path_generator)/launch/one_robot.launch">
  <arg name= "init_pose" value="-x 1 -y 1 -z 0" />
  <arg name= "robot_name" value="robot2" />
  </include> 
</group

>
<!-- Begin robot 3 -->
<group ns ="robot3">
  <param name = "tf_prefix" value ="robot3_tf"/>
  <include file = "$(find path_generator)/launch/one_robot.launch">
  <arg name= "init_pose" value="-x 5 -y 10 -z 0" />
  <arg name= "robot_name" value="robot3" />
  </include> 
</group>

<!-- Begin robot 4 -->
<group ns ="robot4">
  <param name = "tf_prefix" value ="robot4_tf"/>
  <include file = "$(find path_generator)/launch/one_robot.launch">
  <arg name= "init_pose" value="-x 10 -y -20 -z 0" />
  <arg name= "robot_name" value="robot4" />
  </include> 
</group>


</launch>
